src/randoop/operation/MethodCall.java

```diff
@@ -341,18 +359,80 @@ public boolean isStatic() {
   }
 
   /**
-   * A string representing this method's signature. Examples:
-   *
-   * java.util.ArrayList.get(int)
-   * java.util.ArrayList.add(int,java.lang.Object)
+   * {@inheritDoc}
+   * The descriptor for a method is a string representing the method signature.
+   *  
+   * Examples:
+   *  java.util.ArrayList.get(int)
+   *  java.util.ArrayList.add(int,java.lang.Object)
    */
+  @Override
   public String toParseableString() {
-    return Reflection.getSignature(method);
+    return MethodSignatures.getSignatureString(this.method);
   }
 
-  public static Operation parse(String s) {
-    return MethodCall.getRMethod(Reflection.getMethodForSignature(s));
+  /**
+   * Parses a method call in a string descriptor and returns a {@link MethodCall}
+   * object. Should satisfy <code>parse(op.toParseableString()).equals(op)</code>
+   * for Operation op.
+   * @see OperationParser#parse(String)
+   * 
+   * @param s  a string descriptor
+   * @return the {@link MethodCall} object described by the string.
+   * @throws OperationParseException if s does not match expected descriptor.
+   */
+  public static Operation parse(String s) throws OperationParseException {
+    return MethodCall.createMethodCall(MethodSignatures.getMethodForSignatureString(s));
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return the class in which this method is declared.
+   */
+  @Override
+  public Class<?> getDeclaringClass() {
+    return method.getDeclaringClass();
   }
 
+  /**
+   * callsMethodIn determines whether the current method object
+   * calls one of the methods in the list.
+   * @param list method objects to compare against.
+   * @return true if method called by this object is in the given list.
+   */
+  public boolean callsMethodIn(List<Method> list) {
+    return list != null && list.contains(method);
```

Comment:
Why does this test whether the list is null?  It seems that could mask errors, especially since the specification does not say that list is allowed to be null.  I don't see any clients that would actually pass null, but the test strongly suggests to me that they are allowed to do so, despite the lack of documentation about it.  I suggest you shorten the body to just `return list.contains(method);`